packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,163 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex: regex }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   When `prefixRewrite` is provided, `prefixPathMatch` must start and end with the '/' character.
+   *   If provided as "/", matches all requests.
+   *   For example, if your virtual service name is "my-service.local"
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param prefixRewrite Specify either disabling automatic rewrite to '/' or rewriting to specified prefix path.
+   */
+  public static startsWith(prefixPathMatch: string, prefixRewrite?: HttpGatewayRoutePathRewrite): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, prefixRewrite);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly pathRewrite?: HttpGatewayRoutePathRewrite,
+  ) { super(); }
+
+  bind(scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.pathRewrite?.bind(scope).prefixPathRewrite,
+    };
+  }
+}
+
+class HttpGatewayRouteWholePathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly wholePathMatch: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.wholePathMatch,
+      wholePathRewrite: this.wholePathRewrite?.exact
+        ? this.wholePathRewrite
+        : undefined,
```

Comment:
```suggestion
      wholePathRewrite: this.wholePathRewrite,
```

Reply:
Pushing back the suggestion. 
Actually I initially tried this implementation and noticed CloudFormation template would get empty `Path` when the `rewrite` is not specified.